redundancy/redundancy.go

```diff
@@ -39,19 +44,24 @@ func NewNodeRedundancy(arg ArgNodeRedundancy) (*nodeRedundancy, error) {
 	if check.IfNil(arg.ObserverPrivateKey) {
 		return nil, ErrNilObserverPrivateKey
 	}
+	err := common.CheckMaxRoundsOfInactivity(arg.MaxRoundsOfInactivity)
+	if err != nil {
+		return nil, err
+	}
 
 	nr := &nodeRedundancy{
-		redundancyLevel:    arg.RedundancyLevel,
-		messenger:          arg.Messenger,
-		observerPrivateKey: arg.ObserverPrivateKey,
+		handler:               common.NewRedundancyHandler(),
+		maxRoundsOfInactivity: arg.MaxRoundsOfInactivity,
+		messenger:             arg.Messenger,
+		observerPrivateKey:    arg.ObserverPrivateKey,
 	}
 
 	return nr, nil
 }
 
 // IsRedundancyNode returns true if the current instance is used as a redundancy node
 func (nr *nodeRedundancy) IsRedundancyNode() bool {
-	return nr.redundancyLevel != 0
+	return nr.handler.IsRedundancyNode(nr.maxRoundsOfInactivity)
```

Comment:
nr.mutNodeRedundancy.RLock()
	defer nr.mutNodeRedundancy.RUnlock()
?

Reply:
can add but is not needed due to the fact that the handler's pointer won't change and the function IsRedundancyNode does not need a mutex. Will refactor